src/Components/Aspire.Kafka.Consumer/AspireKafkaConsumerExtensions.cs

```diff
@@ -0,0 +1,87 @@
+        => AddKafkaConsumer(builder, DefaultConfigSectionName, configureConsumerBuilder, configureConsumerConfig, connectionName, serviceKey: null);
+
+    /// <summary>
+    /// Registers <see cref="IConsumer{TKey,TValue}"/> as a keyed singleton for the given <paramref name="name"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The name of the component, which is used as the <see cref="ServiceDescriptor.ServiceKey"/> of the service and also to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureConsumerBuilder">A method used for customizing the <see cref="ConsumerBuilder{TKey,TValue}"/>.</param>
+    /// <param name="configureConsumerConfig">An optional method that can be used for customizing the <see cref="ConsumerConfig"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:Kafka:Consumer:{name}" section.</remarks>
+    public static void AddKeyedKafkaConsumer<TKey, TValue>(this IHostApplicationBuilder builder, string name, Action<ConsumerBuilder<TKey, TValue>>? configureConsumerBuilder = null, Action<ConsumerConfig>? configureConsumerConfig = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        AddKafkaConsumer(builder, $"{DefaultConfigSectionName}:{name}", configureConsumerBuilder, configureConsumerConfig, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddKafkaConsumer<TKey, TValue>(
+        IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<ConsumerBuilder<TKey, TValue>>? configureConsumerBuilder,
+        Action<ConsumerConfig>? configureConsumerConfig,
+        string connectionName,
+        object? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var configSection = builder.Configuration.GetSection(configurationSectionName);
+
+        ConsumerConfig config = new();
+        configSection.Bind(config);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            config.BootstrapServers = connectionString;
+        }
+
+        configureConsumerConfig?.Invoke(config);
+
+        ConsumerBuilder<TKey, TValue> CreateConsumerBuilder(IServiceProvider sp)
+        {
+            // Create and configure the consumer builder
+            var consumerBuilder = new ConsumerBuilder<TKey, TValue>(config);
+            configureConsumerBuilder?.Invoke(consumerBuilder);
+
+            return consumerBuilder;
+        }
+
+        if (serviceKey is null)
```

Comment:
Are there any other interesting types to add to the container?

Reply:
I added `ProducerConfig` and `ConsumerConfig` to ease unit testing. Also added `Action<ProducerBuilder<TKey, TValue>>` and `Action<ConsumerBuilder<TKey, TValue>>` to only rely on DI to new the `XXXBuilder<TKey, TValue>` types.